app/controllers/discovered_hosts_controller.rb

```diff
@@ -292,6 +293,21 @@ def find_by_name(*includes)
     @host
   end
 
+  def check_for_subnet
+    hosts_without_subnet = []
+    case params[:action]
+      when 'reboot', 'auto_provision'
+        if @host.subnet.nil?
+          process_warning :warning_msg => _("Discovered host reported from unknown subnet, communication will not be proxied."), :warning_redirect => request.referer
+        end
+      when 'submit_multiple_reboot', 'submit_multiple_auto_provision'
+        hosts_without_subnet = @hosts.select { |host| host.subnet.nil? }.pluck(:name)
```

Comment:
Nitpick: This could render to hundreds of host, limiting this to 5 would be good.

Reply:
sure, good catch.